File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/object/LtrpVehicle.java
package lt.ltrp.object;


import lt.ltrp.VehicleController;
import lt.ltrp.data.FuelTank;
import lt.ltrp.data.VehicleRadio;
import lt.ltrp.data.TaxiFare;
import net.gtaun.shoebill.data.AngledLocation;
import net.gtaun.shoebill.data.Location;
import net.gtaun.shoebill.object.Vehicle;

import java.util.List;

/**
 * @author Bebras
 *         2016.04.12.
 */
public interface LtrpVehicle extends Vehicle, InventoryEntity {


    static final float SPEED_MAGIC_NUMBER = 170f;

    static List<? extends LtrpVehicle> get() {
        return VehicleController.get().getVehicles();
    }

    static LtrpVehicle getById(int id) {
        return VehicleController.get().getById(id);
    }

    static LtrpVehicle getByVehicle(Vehicle vehicle) {
        return VehicleController.get().getByVehicle(vehicle);
    }

    static LtrpVehicle getByUniqueId(int uniqueid) {
        return VehicleController.get().getByUniqueId(uniqueid);
    }

    static LtrpVehicle getClosest(Location loc, float distance) {
        return VehicleController.get().getClosest(loc, distance);
    }

    static LtrpVehicle getClosest(LtrpPlayer player, float distance) {
        return getClosest(player.getLocation(), distance);
    }

    static LtrpVehicle getClosest(Location loc) {
        return getClosest(loc, Float.MAX_VALUE);
    }

    FuelTank getFuelTank();
    void setFuelTank(FuelTank fuelTank);
    AngledLocation getSpawnLocation();
    void setSpawnLocation(AngledLocation spawnLocation);
    boolean isLocked();
    void setLocked(boolean locked);
    float getMileage();
    void setMileage(float mileage);
    int getSpeed();
    String getLicense();
    void setLicense(String license);
    void sendActionMessage(String s, float distance);
    void sendStateMessage(String s, float distance);
    void sendStateMessage(String s);
    void sendActionMessage(String s);
    boolean isUsed();
    VehicleRadio getRadio();
    LtrpPlayer getDriver();
    void setDriver(LtrpPlayer player);
    TaxiFare getTaxi();

}

File: shoebill_libs/LtrpApi/src/main/java/lt/ltrp/object/LtrpVehicle.kt
package lt.ltrp.`object`

import lt.ltrp.data.FuelTank
import net.gtaun.shoebill.data.AngledLocation
import net.gtaun.shoebill.data.Location
import net.gtaun.shoebill.entities.Vehicle

/**
 * @author Bebras
 *         2016.04.12.
 */
interface LtrpVehicle: InventoryEntity {

    companion object {
        val SPEEC_MAGIC_NUMBER = 170.0

        protected val vehicles: MutableList<LtrpVehicle> = mutableListOf()

        fun get(): List<LtrpVehicle> {
            return vehicles
        }

        fun getById(uuid: Int): LtrpVehicle? {
            if(uuid == Entity.INVALID_ID) return null
            return vehicles.firstOrNull { it.UUID == uuid }
        }

        fun getByVehicle(vehicle: Vehicle): LtrpVehicle? {
            return vehicles.firstOrNull { it.vehicle == vehicle }
        }

        fun getClosest(location: Location, distance: Float): LtrpVehicle? {
            return vehicles
                    .filter { it.vehicle != null }
                    .filter { it.vehicle?.location?.distance(location)!! < distance }
                    .minBy { it.vehicle?.location?.distance(location)!! }
        }

        fun getClosest(player: LtrpPlayer, distance: Float) = getClosest(player.player.location, distance)
        fun getClosest(location: Location) = getClosest(location, Float.MAX_VALUE)
    }

    val vehicle: Vehicle?
    var spawnlocation: AngledLocation
    val fuelTank: FuelTank

    var isLocked: Boolean
    var mileage: Float
    val speed: Float
    var license: String
    val isUsed: Boolean
    var driver: LtrpPlayer?
    var getLastDriver: LtrpPlayer?
    var getPullingVehicle: LtrpVehicle?

    fun sendActionMessage(message: String, distance: Float)
    fun sendActionMessage(message: String) = sendActionMessage(message, 10f)

    fun sendStateMessage(message: String, distance: Float)
    fun sendStateMessage(message: String) = sendStateMessage(message, 10f)

    fun isSeatWindowOpen(seatId: Int): Boolean
}
/*
public interface LtrpVehicle extends Vehicle, InventoryEntity {

    VehicleRadio getRadio(); // TODO new module
    TaxiFare getTaxi(); // TODO new module
}
*/